src/site/stages/build/drupal/assets.js

```diff
@@ -11,20 +11,28 @@ const PUBLIC_URLS_NO_SCHEME = Object.entries(PUBLIC_URLS).reduce(
   {},
 );
 
-function replacePathInData(data, replacer) {
+function replacePathInData(data, replacer, ancestors = []) {
+  // Circular references happen when an entity in the CMS has a child entity
+  // which is also in its ancestor tree. When this hapens, this function becomes
+  // infinitely recursive. This check looks for circular references and, when
+  // found, exits early because it's already checked that data.
+  if (ancestors.includes(data)) return data;
```

Comment:
I'm guessing that this also doesn't have to do with unpublished entities, so can you expand the description a bit? If this would cause infinite recursion, how is it just now being noticed?

Reply:
Right, good call. As I continued working through the Metalsmith build, getting it to use data from the CMS export, I ran into a `Maximum call stack size exceeded` error on this function because data from the CMS export contained circular references.

This isn't directly related to unpublished entities, but I needed the work from this branch to get this far. I just committed this here instead of to a separate branch. :sweat_smile: Further changes are finding a home in a separate branch. :+1: